Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -1236,19 +1228,21 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
+    private func handleRestartAdapter() async throws {
+        let tunnelConfiguration = try await generateTunnelConfiguration(
+            serverSelectionMethod: currentServerSelectionMethod,
+            dnsSettings: settings.dnsSettings,
+            regenerateKey: false)
+
+        try await updateTunnelConfiguration(updateMethod: .useConfiguration(tunnelConfiguration),
+                                            reassert: false,
+                                            regenerateKey: false)
+    }
```

Comment:
Convenience method for restarting the adapter very quickly and without going through server re-registration.